libscript/sources/Engine.cxx

```diff
@@ -0,0 +1,77 @@
+#include "pivot/script/Engine.hxx"
+#include "Logger.hpp"
+#include "pivot/script/Exceptions.hxx"
+
+namespace pivot::ecs::script
+{
+
+Engine::Engine(systems::Index &systemIndex, component::Index &componentIndex)
+    : _systemIndex(systemIndex), _componentIndex(componentIndex)
+{
+}
+
+std::string Engine::loadFile(const std::string &file, bool isContent, bool verbose)
+{    // Load a file to register all descriptions in the file, as well as store the trees for the system entry points
+    if (verbose) logger.info("script::Engine") << ": Loading file " << file;
+    // TODO: return string with error on top of the node to deamImgui
+    Node fileNode = _parser.ast_from_file(file, isContent, verbose);    // generate abstract syntax tree from file
+    if (verbose) parser::printFileNode(fileNode);
+
+    // TODO: return string with error on top of the node
+    std::vector<systems::Description> sysDescs = interpreter::registerDeclarations(
+        fileNode, _componentIndex);    // register component descriptions, and retrieve system descriptions
+    for (systems::Description &desc: sysDescs) {
+        desc.system = std::bind(&Engine::systemCallback, this, std::placeholders::_1, std::placeholders::_2,
+                                std::placeholders::_3);    // add the callback to the descriptions
+        _systemIndex.registerSystem(desc);                 // register system in the not-so-global index
+        try {
+            Node systemEntry = getEntryPointFor(
+                desc.name, fileNode);    // only store the entry point for the system, the rest is in the description
+            _systems.insert({desc.name, systemEntry});
+        } catch (const std::exception &e) {
+            return e.what();    // unlikely branch, return error to dearImGui
+        }
```

Comment:
Do not catch the error here. If the error must be caught to display in DearImGui, do it in the editor.
```suggestion
        Node systemEntry = getEntryPointFor(
            desc.name, fileNode);    // only store the entry point for the system, the rest is in the description
        _systems.insert({desc.name, systemEntry});
```